contracts/staking/StakeManager.sol

```diff
@@ -97,8 +99,31 @@ contract StakeManager is Validator, IStakeManager, RootChainable, Lockable {
     _;
   }
 
-  function stake(uint256 amount, address signer, bool isContract) external {
-    stakeFor(msg.sender, amount, signer, isContract);
+  // TopUp heimdall fee
+  function topUpForFee(uint256 validatorId, uint256 heimdallFee) public onlyStaker(validatorId) {
```

Comment:
yeah, I would be happy to but this function can be used to flood Heimdall bridge for validators and using `onlyStaker` makes is trusted function. at least we can ban that `validatorId` in case of spam top-ups.